examples/lock-app/efr32/src/LockManager.cpp

```diff
@@ -407,6 +445,165 @@ bool LockManager::SetCredential(chip::EndpointId endpointId, uint16_t credential
     return true;
 }
 
+DlStatus LockManager::GetWeekdaySchedule(chip::EndpointId endpointId, uint8_t weekdayIndex, uint16_t userIndex,
+                                         EmberAfPluginDoorLockWeekDaySchedule & schedule)
+{
+    if ((weekdayIndex < MIN_SCHEDULE_INDEX) || (weekdayIndex > MAX_SCHEDULE_PER_USER))
+    {
+        ChipLogError(Zcl, "Cannot get schedule - schedule index is out of range [endpoint=%d,index=%d]", endpointId, weekdayIndex);
+        return DlStatus::kFailure;
+    }
+    if ((userIndex < MIN_USER_INDEX) || (userIndex > DOOR_LOCK_MAX_USERS))
+    {
+        ChipLogError(Zcl, "Cannot get schedule - user index is out of range [endpoint=%d,index=%d]", endpointId, userIndex);
+        return DlStatus::kFailure;
+    }
+
+    // doorlock server checks for valid indices
+    uint8_t adjustedWeekDayIndex = weekdayIndex - 1;
+    uint16_t adjustedUserIndex   = userIndex - 1;
+
+    auto & scheduleInStorage = mWeekdaySchedule[adjustedUserIndex][adjustedWeekDayIndex];
```

Comment:
Removed the extra auto, and switched to -- instead of -1. Ended up getting rid of MIN_USER_INDEX as well. These really helped clean up the code, thanks.